src/gui/dialogs/attack_predictions.cpp

```diff
@@ -220,8 +220,9 @@ void attack_predictions::set_data(window& window, const combatant_data& attacker
 	// Time of day modifier.
 	const unit& u = *attacker.unit_;
 
-	const int tod_modifier = combat_modifier(resources::gameboard->units(), resources::gameboard->map(),
+	int unit_combat_modifier = combat_modifier(resources::gameboard->units(), resources::gameboard->map(),
 		u.get_location(), u.alignment(), u.is_fearless());
+	const int tod_modifier = weapon->specials_alignment(unit_combat_modifier, u.is_fearless());
```

Comment:
A lot of the logic of `specials_alignment` and `under_alignment_abilities` seems to duplicate the logic that's already in `combat_modifier` or `generic_combat_modifier`. Instead of passing `unit_combat_modifier` to `specials_alignment`, could these be done as:

```
const auto effective_alignment = weapon->specials_alignment();
const auto tod_modifier = combat_modifier(resources::gameboard->units(), resources::gameboard-->map(),
    u.get_location(), effective_alignment, u.is_fearless());
```
?